crates/bevy_render/src/pass/pass.rs

```diff
@@ -1,23 +1,38 @@
 use super::Operations;
-use crate::{renderer::TextureId, Color};
+use crate::{
+    renderer::{RenderResourceId, SwapChainTextureId, TextureViewId},
+    Color,
+};
 
 #[derive(Debug, Clone)]
 pub enum TextureAttachment {
-    Id(TextureId),
+    View(TextureViewId),
+    SwapChain(SwapChainTextureId),
     Name(String),
     Input(String),
 }
 
-impl TextureAttachment {
-    pub fn get_texture_id(&self) -> Option<TextureId> {
-        if let TextureAttachment::Id(texture_id) = self {
-            Some(*texture_id)
-        } else {
-            None
+impl From<RenderResourceId> for TextureAttachment {
+    fn from(resource_id: RenderResourceId) -> Self {
+        match resource_id {
+            RenderResourceId::Texture(texture_view) => TextureAttachment::View(texture_view),
+            RenderResourceId::SwapChain(texture_id) => TextureAttachment::SwapChain(texture_id),
+            _ => panic!(),
```

Comment:
`panic!("my message")` with a message is usually better than just an explicit `panic!()` so we can better convey the reasoning for panicking.

Reply:
That's my bad.